assets/benchmarks/shaders/Benchmark_Texture.hlsl

```diff
@@ -14,9 +14,112 @@
+[[vk::combinedImageSampler]]
+SamplerState yuvsampler6 : register(s18);
+[[vk::combinedImageSampler]]
+Texture2D YUVTex7 : register(t19);
+[[vk::combinedImageSampler]]
+SamplerState yuvsampler7 : register(s19);
+[[vk::combinedImageSampler]]
+Texture2D YUVTex8 : register(t20);
+[[vk::combinedImageSampler]]
+SamplerState yuvsampler8 : register(s20);
+[[vk::combinedImageSampler]]
+Texture2D YUVTex9 : register(t21);
+[[vk::combinedImageSampler]]
+SamplerState yuvsampler9 : register(s21);
+
+float rand_float(uint seed)
+{
+    const uint state = seed * 747796405u + 2891336453u;
+    const uint word = ((state >> ((state >> 28u) + 4u)) ^ state) * 277803737u;
+    //#define FAKE_HASH
+    #if defined(FAKE_HASH)
+    return ((word >> 22u) ^ word) * (1.0 / 4.0);
+    #else
+    return ((word >> 22u) ^ word) * (1.0 / 4294967296.0);
+    #endif
+}
+
 
 float4 psmain(VSOutputPos input) : SV_TARGET
 {
-    return Tex0.Load(uint3(input.position.x, input.position.y, /* mipmap */ 0));
+    float4 c0 = Tex0.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c1 = Tex1.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c2 = Tex2.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c3 = Tex3.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c4 = Tex4.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c5 = Tex5.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c6 = Tex6.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c7 = Tex7.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c8 = Tex8.SampleLevel(pointsampler, input.texcoord, 0);
+    float4 c9 = Tex9.SampleLevel(pointsampler, input.texcoord, 0);
+
+    float4 sum_c = (c0 + c1  + c2  + c3/* + c4 + c5 + c6+ c7 + c8 + c9*/)/6.f ;
+
+    float4 cYUV0 = YUVTex0.SampleLevel(yuvsampler0, input.texcoord, 0);
+    float4 cYUV1 = YUVTex1.SampleLevel(yuvsampler1, input.texcoord, 0);
+    float4 cYUV2 = YUVTex2.SampleLevel(yuvsampler2, input.texcoord, 0);
+    float4 cYUV3 = YUVTex3.SampleLevel(yuvsampler3, input.texcoord, 0);
+    float4 cYUV4 = YUVTex4.SampleLevel(yuvsampler4, input.texcoord, 0);
+    float4 cYUV5 = YUVTex5.SampleLevel(yuvsampler5, input.texcoord, 0);
+    float4 cYUV6 = YUVTex6.SampleLevel(yuvsampler6, input.texcoord, 0);
+    float4 cYUV7 = YUVTex7.SampleLevel(yuvsampler7, input.texcoord, 0);
+    float4 cYUV8 = YUVTex8.SampleLevel(yuvsampler8, input.texcoord, 0);
+    float4 cYUV9 = YUVTex9.SampleLevel(yuvsampler9, input.texcoord, 0);
+
+    float4 sum_c_yuv = (cYUV0/* + cYUV1  + cYUV2  + cYUV3  + cYUV4 + cYUV5 + cYUV6  + cYUV7  + cYUV8  + cYUV9*/)/1.f ;
+
+    const uint rd_w = 2664;
+    const uint pixel_id = uint(input.position.y * rd_w + input.position.x);
+    const float red = rand_float(pixel_id);
+    const float green = rand_float(pixel_id + 1);
+    const float blue = rand_float(pixel_id - 1);
+    float4 noise = float4(red, green, blue, 1.0f);
+
+    sum_c = sum_c;// * 0.5f + sum_c_yuv *0.5f;// * 0.1f + noise*0.8f;
```

Comment:
@wangra-google do we actually need to generate the noise if we aren't using it?

Reply:
ideally we need to have separate shaders for the noise case. I leave the code here for quick testing.